include/dr/details/normal_distributed_iterator.hpp

```diff
@@ -62,6 +60,7 @@ class normal_distributed_iterator_accessor {
           std::min(-offset, difference_type(idx_) + 1);
```

Comment:
@BenBrock The `normal_distributed_iterator` assumes that it can join the individual segments into an iterator space for the distributed data structure, and the segments can be any length. This would not work well for interleave with a block size of 1. We would need $n$ segments to describe it. For cyclic, there should only be 1 segment per rank, and we need a `cyclic_distributed_iterator` that knows how to map the distributed iterator space to a segment and offset.

Reply:
A related issue is that MHP can't handle a zip where a segment contains data in multiple ranks. If I know that the segments are not single rank, then I can use the fallback--direct use of the distributed iterator/range. Is there some way to mark this so I do not have to check the segments at runtime? Currently, I use an empty segments range as a marker.